codice/alliance
# CAL-717 Added post query plugin to audit classified info

#### What does this PR do?
Added a new PostQuery Plugin to audit classified/controlled metacards. Currently is it set to monitor Classification, Releasability, DisseminationControls, and Codewords attributes based on user configured values. Documentation was added on how to configure the new plugin. The current audit message only logs the metacard's ID. 

#### Who is reviewing it? 
@austinsteffes 
@ahoffer 

#### Choose 2 committers to review/merge the PR.
@brjeter
@ricklarsen - Documentation
@stustison

#### How should this be tested?
1. Standard Install with no issues
2. verify new documentation looks good
3. See `Audit Controlled Access Plugin` aka `security-auditcontrolledaccess` is not installed by default 
4. Verify you can install with no issues and feature ends up located under the `IC Security` application
4. Ingest a file with above mentioned security attributes 
5. Query and verify no issues or the specific logs to the "security.log" file
6. Configure the `Audit Controlled Access Plugin` to audit the specific values in the ingested file
7. Query again and verify no issues
8. Verify that now there are audits in the "security.log" file saying `The classified metacard with id {metacard.Id} is being returned.`

See me for good test data to ingest, questions, and more info if necessary.

#### What are the relevant tickets?
Fixes: #717

#### Checklist:
